### PR TITLE
fix(renderer): fix conditional rendering issue

### PR DESCRIPTION
**Pull request by yigityuce**
fix conditional rendering issue by applying to find actual DOM element(s) when about the relocate element is slot ref

fixes one of the issues raised in #5335

## What is the current behavior?

Slot elements are not rendered conditionally if they are not wrapped with builtin element such as div.

GitHub Issue Number: #5335 

## What is the new behavior?

Fixed

## Does this introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: src/runtime/vdom/vdom-render.ts
Comment (christian-bromann) [inline src/runtime/vdom/vdom-render.ts:218]: Let's put this behind the `experimentalSlotFixes` bug to avoid chances that this impacts behavior of current users that are not opt-into this.
```suggestion
  if (parentElm['s-sr'] && BUILD.experimentalSlotFixes) {
    let node = parentElm;
    while ((node = node.nextSibling as d.RenderNode)) {
      if (node && node['s-sn'] === parentElm['s-sn'] && node['s-sh'] === hostTagName) {
        oldSlotChildNodes.push(node);
      }
    }
  }
```

**Review by christian-bromann — CHANGES_REQUESTED** [review]
Thanks for raising a PR @yigityuce 🙏 

Can I ask you do to add an end-to-end test for this to ensure we don't regress on this? I recommend to look into `test/karma/test-app` and add a new slot-XXX folder and copy your example from the reproduction case.

Let me know if you have any questions or if I can assist you.

File: src/runtime/vdom/vdom-render.ts
Comment (yigityuce) [reply src/runtime/vdom/vdom-render.ts]: done 🕺

**Review by christian-bromann — APPROVED** [review]
Perfect 👌 this LGTM 👍

Let's have someone else from the @ionic-team/stencil review this.

**Comment by yigityuce** [comment]
hey @christian-bromann & @rwaskiewicz  do I need to do sth else to move forward?

**Comment by rwaskiewicz** [comment]
@yigityuce Nope 🙂 I want to land some other infrastructure work to test this out a little more with the Ionic Framework before we land this, but plan on taking a look at the PR itself later today

**Review by rwaskiewicz — APPROVED** [review]
Looks good to me! Thanks for your patience!

**Comment by rwaskiewicz** [comment]
I need to create a PR for the Stencil Site for this PR, I'll merge this once the docs-PR is approved

**Comment by yigityuce** [comment]
hey @rwaskiewicz thanks for the update! If I can do anything to help for that docs side, please let me know

**Pull request merged**
